File: String Matching/Q1. Repeated Substring Pattern/repeatedSubstringPattern.js
/**
 * @param {string} s
 * @return {boolean}
 */
var repeatedSubstringPattern = function (s) {
    let mid = Math.floor(s.length / 2)
    const len = s.length

    for (let i = 1; i <= mid; i++) {
        let sub = s.substr(0, i)
        let rep = Math.floor(len / sub.length)

        if (sub.repeat(rep) === s) {
            return true
        }
    }
    return false
};

File: String Matching/Q1. Repeated Substring Pattern/repeatedSubstringPattern.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    bool repeatedSubstringPattern(string s) {
        int sz = s.size();
        int mid = sz/2;

        for(int i=1; i<=mid; i++){
            string sub = s.substr(0,i);
            int rep = sz / sub.size();
            string form = "";

            //repeat
            while(rep > 0){
                form+=sub;
                rep--;
            }
            if(form == s)return true;
        }

        return false;
    }
};